vtex-apps/store-components
# Migrate Newsletter to TypeScript and add support for 'name' argument

#### What problem is this solving?

This component was still written in JS. Migrating it to TypeScript should make it easier for users to use it and for us to maintain.

Also, this enables users to ask for the names of clients subscribing to their newsletter. 

#### How to test it?

Go to this [Workspace](https://victormiranda--storecomponents.myvtex.com/) and signup for the newsletter.

Also, you can go to [Site Editor](https://victormiranda--storecomponents.myvtex.com/admin/cms/site-editor) and see the props exposed there.

#### Screenshots or example usage:

![Screen Shot 2020-10-05 at 16 26 43](https://user-images.githubusercontent.com/27777263/95122917-916eaa80-0727-11eb-9dd3-242b6331d7b8.png)

#### Depends on

- https://github.com/vtex-apps/store-graphql/pull/500

#### How does this PR make you feel? [:link:](http://giphy.com/)



![](put .gif link here - can be found under "advanced" on giphy)
